kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/ssl/KubernetesClientSslBuilder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.ssl;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.io.ResourceResolver;
+import io.micronaut.http.client.netty.ssl.NettyClientSslBuilder;
+import io.micronaut.http.ssl.ClientSslConfiguration;
+import io.micronaut.http.ssl.SslConfiguration;
+import io.micronaut.kubernetes.client.openapi.config.model.AuthInfo;
+import io.micronaut.kubernetes.client.openapi.config.model.Cluster;
+import io.micronaut.kubernetes.client.openapi.config.KubeConfig;
+
+import java.io.ByteArrayInputStream;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * The ssl builder which uses data from a kube config file to create a key store and trust store.
+ */
+@Internal
+public class KubernetesClientSslBuilder extends NettyClientSslBuilder {
```

Comment:
does this need to be `public`? can it be `final`?

Reply:
added `final`